### ISSUE TITLE
Insert Link: A href link gets corrupted when using the code editor

### ISSUE THREAD

**Issue by kwinz** [WordPress/gutenberg#9606]
**Describe the bug**
A URL in a link is automatically converted from `&` to `&amp;` when editing a post in the Gutenberg code editor. This breaks the link.
Since Links regularly contain `&` and every edit can cause a subtile bug I consider this a severe bug in Gutenberg.

**To Reproduce**

1. Open a Post.
2. Goto code editor (CTRL)+(SHIFT)+(ALT)+(M)
3. Edit the post.
4. Publish
5. A URL in a link (e.g `href="example.com?id=1&example=true"`) is automatically converted from `&` to `&amp;` (e.g `href="example.com?id=1&amp;example=true"`)
6. This breaks the link.
7. This happens especially when posting a link to a timestamped youtube video.

Tested with Version 3.7.0 | By Gutenberg,
WordPress. Version 4.9.8

**Expected behavior**
`&` should never be substituted with `&amp;` when using the code editor, especially not within URLs.

**Comment by kwinz** [comment]
Okay, thanks @liamdigitalstorm. I will try to reproduce it in another WordPress installation without extra plugins and provide a minimal post-example to reproduce for you.

**Comment by designsimply** [comment]
Using `&amp;` is the traditionally valid way of writing ampersands in attributes in HTML. Here is some information about that from the W3C guidelines: https://www.w3.org/TR/xhtml1/guidelines.html#C_12

> For example, when the `href` attribute of the `a` element refers to a CGI script that takes parameters, it must be expressed as `http://my.site.dom/cgi-bin/myscript.pl?class=guest&amp;name=user` rather than as `http://my.site.dom/cgi-bin/myscript.pl?class=guest&name=user`.

You can test this:

1. Go to https://validator.w3.org/#validate_by_input
1. Add one of the HTML examples from below.
1. Click the "Check" button.

```
<!DOCTYPE html>
<html lang="en-US">
<head>
<title>test</title>
</head>
<body>
<a href="http://example.com?id=1&amp;example=true">link</a>
</body>
</html>
```

or 

```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en-US">
<head>
<title>test</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
</head>
<body>
<p><a href="http://example.com?id=1&amp;example=true">link</a></p>
</body>
</html>
```

**Comment by designsimply** [comment]
I tested with WordPress 4.9.8 and Gutenberg 3.7 and I was unable to see the link break (step 6).

In my test, I used Firefox 62.0 on macOS 10.13.6 with the Twenty Seventeen theme active and added the following link inside a paragraph block while using the code editor:

```
<a href="https://www.youtube.com/watch?v=h--fvyPu5e4&feature=youtu.be&t=15s">link to a timestamped youtube video</a>
```

May I ask what OS and browser versions you are using? For reference, may I also ask what theme you are testing with? Also, where/how did you copy/paste the link from before you pasted it into the Gutenberg code editor (e.g. directly from the browser title bar or from another source such as a Google Doc or another website)?

**Comment by kwinz** [comment]
Dear @designsimply:

Short answer: Thank you for trying to reporduce! I tried it in a different browser and it did not break the link any more. I think this issue can be closed.

More info:
1. As you suspected the link breakage was caused by my browser or more specifically an extension, that I had synced on all my chrome instances on my computers.
2. Thanks for researching the correct href behaviour. I also found https://stackoverflow.com/questions/3705591/do-i-encode-ampersands-in-a-href
3. The current behavior seems to be that if I copy paste `https://www.youtube.com/watch?v=tuDbUg-FOMY&feature=youtu.be&t=713` into href="<paste here>" in the code editor it gets automatically converted to: `href="https://www.youtube.com/watch?v=tuDbUg-FOMY&amp;feature=youtu.be&amp;t=713"`. But both in the visual editor, *and* in the source code of the finished published article in the viewer's browser the article the ampersand is not encoded as &amp;
